cgltf_write.h

```diff
@@ -590,6 +596,22 @@ static void cgltf_write_material(cgltf_write_context* context, const cgltf_mater
 			cgltf_write_floatprop(context, "transmissionFactor", params->transmission_factor, 0.0f);
 			cgltf_write_line(context, "}");
 		}
+		if (material->has_volume)
+		{
+			const cgltf_volume* params = &material->volume;
+			cgltf_write_line(context, "\"KHR_materials_volume\": {");
+			CGLTF_WRITE_TEXTURE_INFO("thicknessTexture", params->thickness_texture);
+			cgltf_write_floatprop(context, "thicknessFactor", params->thickness_factor, 0.0f);
+			if (cgltf_check_floatarray(params->attenuation_color, 3, 1.0f))
+			{
+				cgltf_write_floatarrayprop(context, "attenuationColor", params->attenuation_color, 3);
+			}
+			if (params->attenuation_distance < FLT_MAX) 
```

Comment:
I would suggest that if `params->attenuation_distance` is out-of-range at all, it should not be written.  In other words test for > 0.0 && < FLT_MAX.

Reply:
The thought here is that a value of 0.0 for `attenuation_distance` is not allowed by the spec, as a value of 0.0 implies that this material is opaque.  I don't see that level of validation done elsewhere in the writer, so I'm assuming that this logic is handled outside this library at the application level.

Reply:
Ok, fair enough.